src/pybind/mgr/dashboard_v2/controllers/osd.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import json
+
+from mgr_module import CommandResult
+from ..tools import ApiController, AuthRequired, RESTController
+from .. import logger
+
+
+@ApiController('osd')
+@AuthRequired()
+class Osd(RESTController):
+    def get_counter(self, daemon_name, stat):
+        return self.mgr.get_counter('osd', daemon_name, stat)[stat]
+
+    def get_rate(self, daemon_name, stat):
+        data = self.get_counter(daemon_name, stat)
+        rate = 0
+        if data and len(data) > 1:
+            rate = (data[-1][1] - data[-2][1]) / float(data[-1][0] - data[-2][0])
+        return rate
+
+    def get_latest(self, daemon_name, stat):
+        data = self.get_counter(daemon_name, stat)
+        latest = 0
+        if data and data[-1] and len(data[-1]) == 2:
+            latest = data[-1][1]
+        return latest
+
+    def list(self):
+        osds = self.get_osd_map()
+        # Extending by osd stats information
+        for s in self.mgr.get('osd_stats')['osd_stats']:
+            osds[str(s['osd'])].update({'osd_stats': s})
+        # Extending by osd node information
+        nodes = self.mgr.get('osd_map_tree')['nodes']
+        osd_tree = [(str(o['id']), o) for o in nodes if o['id'] >= 0]
+        for o in osd_tree:
+            osds[o[0]].update({'tree': o[1]})
+        # Extending by osd parent node information
+        hosts = [(h['name'], h) for h in nodes if h['id'] < 0]
+        for h in hosts:
+            for o_id in h[1]['children']:
+                if o_id >= 0:
+                    osds[str(o_id)]['host'] = h[1]
+        # Extending by osd histogram data
+        for id in osds:
```

Comment:
you need to you other variable name. `id` will trigger a lint warning. Maybe call it `osd_id`